workspaces/adventure-pack/goodies/java/src/traverse_post_order/AP.java

```diff
@@ -0,0 +1,63 @@
+package traverse_post_order;
+
+import iterable_stream.IterableStream;
+import java.util.NoSuchElementException;
+import java.util.Stack;
+import pair.Pair;
+import simple_iterator.SimpleIterator;
+
+public final class AP {
+
+  /**
+   * Get {@link IterableStream iterable stream} of all the {@link TreeNode nodes} in the binary tree
+   * in post-order.
+   * @param root Binary tree root {@link TreeNode node}.
+   * @return The {@link IterableStream iterable stream} in post-order.
+   */
+  public static IterableStream<TreeNode> traversePostOrder(
+    final TreeNode root
+  ) {
+    final Stack<Pair<TreeNode, Boolean>> stack = new Stack<
+      Pair<TreeNode, Boolean>
+    >();
+    stack.push(new Pair<>(root, false));
+    return IterableStream.from(
+      SimpleIterator.toIterator(() -> {
+        while (!stack.isEmpty()) {
+          Pair<TreeNode, Boolean> peek = stack.pop();
+          TreeNode node = peek.first();
+          boolean didTraverseChildren = peek.second();
```

Comment:
small nit: this variable is not used in the branch when `node == null`, so we should move it closer to where it's used (i.e. below the `if` statement)

And as an aside, lines 28-29 are another opportunity to save some characters with `var`.

Reply:
hey, here I noticed, "didTraverseChildren" variable is defined and used only once at another instance. Hence, I've written the if condition directly as,
`if (nodeAndDidTraverseChildren.second()) {
   return node
}`
This way it eliminates the above said issue and also consistent with the code in inorder-traversal goody for java.